src/ir/datalog/program.h

```diff
@@ -176,6 +197,13 @@ class Program {
   const std::vector<Rule>& rules() const { return rules_; }
   const std::vector<std::string>& outputs() const { return outputs_; }
 
+  bool operator==(const Program& otherProgram) const {
+    return this->relation_declarations_ == 
+      otherProgram.relation_declarations() &&
+      this->rules_ == otherProgram.rules() &&
+      this->outputs_ == otherProgram.outputs();
```

Comment:
Use fields directly. (Can you make sure you change other places also?)
```suggestion
    return this->relation_declarations_ == 
      otherProgram.relation_declarations_ &&
      this->rules_ == otherProgram.rules_ &&
      this->outputs_ == otherProgram.outputs_;
```

Reply:
fixed